### PR TITLE
refactor(shared-data): cleanup modules V3 schema and definitions and add tests

### PR DESCRIPTION
**Pull request by sanni-t**
# Overview

Does cleanup and adds fixes and tests for modules' V3 definitions and schema to get the definitions ready for #9967 

# Changelog

- added js & python tests for checking schema and loading definitions
- made corrections to schema
- made small corrections to V3 module definitions
- fleshed out `ModuleDefinitionV3` type

# Review requests

- make sure changes look good, specially the labwareOffset changes for MagneticModuleV2 and changes in `dev_types`
- did I miss anything? I was going to address the todo to change the typedicts to pydantic models but punting it to a later ticket in order to quickly unblock work for #9967

# Risk assessment

Medium. V3 defs are already being used by the client and protocol engine. This PR makes only small corrections to schemas but it is important to get them right before the api is switched over to using them

### CODE REVIEW

**Review by SyntaxColoring — APPROVED** [review]
Reviewed in a call with @sanni-t. Looks good to me.

The `magneticModuleV2` offset changes bring the schema v3 definition in line with the schema v2 definition, so they make sense to me. My understanding (from @sanni-t) is that no one knows where the incorrect value of `0.3` came from, but it's presumed to be some sort of copy-paste error.

File: shared-data/js/__tests__/moduleSpecsSchema.test.ts [hunk 1]
```diff
@@ -39,8 +46,21 @@ describe('validate all module specs with schema', () => {
...
+
+      expect(validationErrors).toBe(null)
+      expect(valid).toBe(true)
+    })
+  })
+
   it('validate each module specs model matches its filename', () => {
-    MODULE_PATHS.forEach(modulePath => {
+    V2_MODULE_PATHS.forEach(modulePath => {
       const filename = path.parse(modulePath).name
```
Comment (shlokamin) [inline shared-data/js/__tests__/moduleSpecsSchema.test.ts]: IMO if we're doing file name asserts for v2 defs we should also do them for v3 defs too

**Review by shlokamin — APPROVED** [review]
small comment in a test but JS changes LGTM!

**Review by shlokamin — APPROVED** [review]
😄

**Pull request merged**